fre/make/tests/null_example/test_checkout_null_yaml.py

```diff
@@ -0,0 +1,45 @@
+#tests for the create-checkout step of fre-make, for null_model.yaml
+import os
+from fre import fre
+from pathlib import Path
+from fre.pp import configure_script_yaml as csy
+from click.testing import CliRunner
+import subprocess
+from fre.make import createCheckout
+
+# Set example yaml paths, input directory
+test_dir = Path("fre/make/tests/null_example")
+yamlfile = Path(f"{test_dir}/null_model.yaml")
+
+#set platform and target
+platform = "ncrc5.intel"
+target = "debug"
+
+#set output directory
+#out_dir = Path(f"fre/make/tests/null_example/fre_make_out")
+#Path(out_dir).mkdir(parents=True,exist_ok=True)
+
+# Set home for ~/cylc-src location in script
+#os.environ["HOME"]=str(Path(f"{out_dir}"))
+HOME_DIR = os.environ["HOME"]
```

Comment:
So through doing this, we've found it definitely effects other tests that use `HOME`. 

1) Let's move re-setting `HOME` to inside each test that needs it AND make sure to save the default `HOME`  location first. We can look at how Ryan maneuvered it:
define home: https://github.com/NOAA-GFDL/fre-cli/blob/3c9cacb99495a29032ef0f81e0d375963134af01/fre/make/tests/test_create_makefile.py#L54
set home: https://github.com/NOAA-GFDL/fre-cli/blob/3c9cacb99495a29032ef0f81e0d375963134af01/fre/make/tests/test_create_makefile.py#L55
reset home: https://github.com/NOAA-GFDL/fre-cli/blob/3c9cacb99495a29032ef0f81e0d375963134af01/fre/make/tests/test_create_makefile.py#L64